keras_retinanet/bin/debug.py

```diff
@@ -219,6 +231,9 @@ def main(args=None):
     # create the generator
     generator = create_generator(args)
 
+    # sorting list of images for cycling through dataset
+    generator.image_names = sorted(generator.image_names)
```

Comment:
What is the reason for sorting it?

Reply:
Initially I experienced that the images would cycle through in a random order and so pressing the back key would take me to a random image. It might've been an issue with an earlier implementation of how I modified the i index. 

Turns out there is no need to sort it after all and I can cycle through the images sequentially now, as expected.